orangecontrib/text/widgets/owsentimentanalysis.py

```diff
@@ -0,0 +1,95 @@
+import numpy as np
+from nltk.corpus import opinion_lexicon
+from PyQt5.QtWidgets import QApplication
+
+from Orange.data import Domain, ContinuousVariable
+from Orange.widgets import gui
+from Orange.widgets.settings import Setting
+from Orange.widgets.widget import OWWidget
+from orangecontrib.text import Corpus
+from orangecontrib.text.preprocess import WordPunctTokenizer
+
+
+class OWSentimentAnalysis(OWWidget):
+    name = "Sentiment Analysis"
+    description = "Predict sentiment from text."
+    icon = "icons/SentimentAnalysis"
+    priority = 30001
+
+    inputs = [("Corpus", Corpus, "set_corpus")]
+    outputs = [("Corpus", Corpus, )]
```

Comment:
Use new-style definitions.

    class Inputs:
        corpus = Input("Corpus", Corpus)
 
    class Outputs:
        corpus = Output("Corpus", Corpus)

+ see additional the changes below